S1_NRB/metadata/extract.py

```diff
@@ -751,6 +773,54 @@ def calc_pslr_islr(annotation_dict, decimals=2):
     return pslr, islr
 
 
+def calc_wm_ref_stats(wm_ref_files, epsg, bounds):
+    """
+    Calculates the mean wind model reference speed and direction for the wind model annotation layer.
+    
+    Parameters
+    ----------
+    wm_ref_files: list[str]
+        List of paths pointing to the wind model reference files.
+    epsg: int
+        The EPSG code of the current MGRS tile.
+    bounds: list[float]
+        The bounds of the current MGRS tile.
+    
+    Returns
+    -------
+    tuple[float]
+        a tuple with the following values:
+        
+        - ref_speed_mean: Mean wind model reference speed.
+        - ref_direction_mean: Mean wind model reference direction.
+    """
+    files_speed = [f for f in wm_ref_files if f.endswith('Speed.tif')]
+    files_direction = [f for f in wm_ref_files if f.endswith('Direction.tif')]
+    
+    ref_speed = tempfile.NamedTemporaryFile(suffix='.tif').name
+    ref_direction = tempfile.NamedTemporaryFile(suffix='.tif').name
```

Comment:
Please delete these two files once done.

Reply:
Good point. I thought they are deleted anyway since the parameters `delete` and `delete_on_close` are by default True for [NamedTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile). But I guess they need to be properly closed.

There are a few other cases where we use `NamedTemporaryFile`, which I can also correct. I actually got this example from [here](https://github.com/SAR-ARD/S1_NRB/blob/1c00554f22c99deffae8550a968523b110fd98d8/S1_NRB/ard.py#L1081C17-L1081C17).

Reply:
https://github.com/SAR-ARD/S1_NRB/pull/166/commits/cd139fa0551da82bad98b8d2852e3086335c6ebe

Reply:
Mh. perhaps you're right. It would make sense. I just remember a case in `spatialist.Raster` where temporary files created [here](https://github.com/johntruckenbrodt/spatialist/blob/12f8fd3f919c69773c12be9a073f34c28a78db76/spatialist/raster.py#L356) were not deleted and so I manually deleted them [here](https://github.com/johntruckenbrodt/spatialist/blob/12f8fd3f919c69773c12be9a073f34c28a78db76/spatialist/raster.py#L612). Well, now we definitely make sure. Thanks.

Reply:
True, I also didn't delete the files in the example you pointed out. Let's just keep the rest like it is. This is very minor and can be fixed at some later point if necessary.